development/contracts/mixin/ForkAwareStructureOnL2.sol

```diff
@@ -0,0 +1,20 @@
+pragma solidity ^0.8.17;
+
+contract ForkAwareStructureOnL2 {
+    uint256 public chainId;
+
+    modifier onlyFirstTxAfterFork() {
```

Comment:
Just on the naming I would have gone with something like `initializeChain` then the modifers would be like `onlyChainInitialized` and `onlyChainUninitialized`.